src/dyn_colors.rs

```diff
@@ -10,6 +10,7 @@ use core::fmt;
 /// allowing for multiple types of colors to be used at runtime.
 #[allow(missing_docs)]
 #[derive(Copy, Clone, PartialEq, Eq, Debug)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
```

Comment:
Will also want `kebab-case` here.

Trying to figure out what enum representation makes the most sense. Externally tagged is probably OK.